Mathlib/NumberTheory/NumberField/Units.lean

```diff
@@ -149,4 +161,398 @@ theorem rootsOfUnity_eq_torsion [NumberField K] :
 
 end torsion
 
+namespace dirichlet
```

Comment:
Should this maybe even be
```suggestion
namespace dirichletUnitTheorem
```
because I can imagine several topics that could go in the `dirichlet` namespace.

I assume most results in this namespace will be quite specific to the proof. If someone finds a use for one of the results more generally, that is in indication that it should be moved out of this namespace to a different place.